frontend/src/views/GameDetails.vue

```diff
@@ -0,0 +1,249 @@
+<script setup lang="ts">
+import EmptyGame from "@/components/common/EmptyGame.vue";
+import Cover from "@/components/Game/Card/Base.vue";
+import ActionBar from "@/components/Game/Details/ActionBar.vue";
+import AdditionalContent from "@/components/Game/Details/AdditionalContent.vue";
+import BackgroundHeader from "@/components/Game/Details/BackgroundHeader.vue";
+import Info from "@/components/Game/Details/Info/Base.vue";
+import Notes from "@/components/Game/Details/Notes.vue";
+import RelatedGames from "@/components/Game/Details/RelatedGames.vue";
+import Saves from "@/components/Game/Details/Saves.vue";
+import States from "@/components/Game/Details/States.vue";
+import TitleInfo from "@/components/Game/Details/Title.vue";
+import platformApi from "@/services/api/platform";
+import romApi from "@/services/api/rom";
+import storeDownload from "@/stores/download";
+import type { Platform } from "@/stores/platforms";
+import type { DetailedRom } from "@/stores/roms";
+import type { Events } from "@/types/emitter";
+import type { Emitter } from "mitt";
+import { inject, onBeforeMount, ref, watch } from "vue";
+import { useRoute } from "vue-router";
+import { useDisplay } from "vuetify";
+
+// Props
+const route = useRoute();
+const rom = ref<DetailedRom>();
+const platform = ref<Platform>();
+const tab = ref<
+  | "details"
+  | "saves"
+  | "states"
+  | "notes"
+  | "additionalcontent"
+  | "screenshots"
+  | "relatedgames"
+>("details");
+const { smAndDown, mdAndDown, mdAndUp, lgAndUp } = useDisplay();
+const emitter = inject<Emitter<Events>>("emitter");
+const noRomError = ref(false);
+
+// Functions
+async function fetchDetails() {
+  if (!route.params.rom) return;
+
+  await romApi
+    .getRom({ romId: parseInt(route.params.rom as string) })
+    .then((response) => {
+      rom.value = response.data;
+    })
+    .catch((error) => {
+      console.log(error);
+      noRomError.value = true;
+    })
+    .finally(() => {
+      emitter?.emit("showLoadingDialog", { loading: false, scrim: false });
+    });
+
+  if (!noRomError.value) {
+    await platformApi
+      .getPlatform(rom.value?.platform_id)
+      .then((response) => {
+        platform.value = response.data;
+      })
+      .catch((error) => {
+        console.log(error);
+      })
+      .finally(() => {
+        emitter?.emit("showLoadingDialog", { loading: false, scrim: false });
+      });
+  }
+}
+
+onBeforeMount(async () => {
+  emitter?.emit("showLoadingDialog", { loading: true, scrim: false });
+  if (rom.value) {
+    emitter?.emit("showLoadingDialog", { loading: false, scrim: false });
+  } else {
+    await fetchDetails();
+  }
+  const downloadStore = storeDownload();
+  downloadStore.clear();
```

Comment:
this download store clearing, was it copied from the old file?

Reply:
hmm yeah not sure about this one, i'll check!

Reply:
okay it's needed to clear the multipart selector